public/app/plugins/panel/state-timeline/StateTimelineTooltip2.tsx

```diff
@@ -1,102 +1,57 @@
-import { css } from '@emotion/css';
 import React from 'react';
 
-import {
-  arrayUtils,
-  DataFrame,
-  Field,
-  FieldType,
-  getDisplayProcessor,
-  getFieldDisplayName,
-  GrafanaTheme2,
-  LinkModel,
-  TimeRange,
-  TimeZone,
-} from '@grafana/data';
+import { Field, FieldType, getFieldDisplayName, LinkModel, TimeRange } from '@grafana/data';
 import { SortOrder } from '@grafana/schema/dist/esm/common/common.gen';
-import { TooltipDisplayMode, useStyles2, useTheme2 } from '@grafana/ui';
+import { TooltipDisplayMode, useStyles2 } from '@grafana/ui';
 import { VizTooltipContent } from '@grafana/ui/src/components/VizTooltip/VizTooltipContent';
 import { VizTooltipFooter } from '@grafana/ui/src/components/VizTooltip/VizTooltipFooter';
 import { VizTooltipHeader } from '@grafana/ui/src/components/VizTooltip/VizTooltipHeader';
-import { ColorIndicator, ColorPlacement, LabelValue } from '@grafana/ui/src/components/VizTooltip/types';
+import { LabelValue } from '@grafana/ui/src/components/VizTooltip/types';
 import { findNextStateIndex, fmtDuration } from 'app/core/components/TimelineChart/utils';
 
 import { getDataLinks } from '../status-history/utils';
+import { TimeSeriesTooltipProps, getContentItems, getStyles } from '../timeseries/TimeSeriesTooltip';
 
-interface StateTimelineTooltip2Props {
-  data: DataFrame[];
-  alignedData: DataFrame;
-  dataIdxs: Array<number | null>;
-  seriesIdx: number | null | undefined;
-  isPinned: boolean;
-  timeZone?: TimeZone;
+interface StateTimelineTooltip2Props extends TimeSeriesTooltipProps {
   timeRange: TimeRange;
-  mode?: TooltipDisplayMode;
-  sortOrder?: SortOrder;
-  annotate?: () => void;
+  withDuration: boolean;
 }
 
 export const StateTimelineTooltip2 = ({
-  data,
-  alignedData,
+  frames,
+  seriesFrame,
   dataIdxs,
   seriesIdx,
-  timeZone,
-  timeRange,
   mode = TooltipDisplayMode.Single,
   sortOrder = SortOrder.None,
+  scrollable = false,
   isPinned,
   annotate,
+  timeRange,
+  withDuration,
 }: StateTimelineTooltip2Props) => {
   const styles = useStyles2(getStyles);
-  const theme = useTheme2();
 
-  const datapointIdx = seriesIdx != null ? dataIdxs[seriesIdx] : dataIdxs.find((idx) => idx != null);
+  const xField = seriesFrame.fields[0];
```

Comment:
in theory, the `seriesFrame` (which is `alignedFrame`) param of the wrapping render prop in the parent _should_ guarantee this for us. it is generated in `StateTimelinePanel > TimelineChart > GraphNG  > preparePlotFrame()`, and uPlot uses this `alignedFrame` to grab data for rendering, which requires the x field to be first in non-mode-2 viz panels.

this is yet another case of needing to move the data prep up waaay up into the panel's prepareGraphableFields-equivalent util function so that the logic was super clear rather than :exploding_head:. we will be refactoring this  soon, but def not in this PR :sweat_smile: